src/main/java/forestry/plugins/PluginTechReborn.java

```diff
@@ -0,0 +1,136 @@
+import forestry.core.config.Constants;
+import forestry.core.fluids.Fluids;
+import forestry.core.items.EnumElectronTube;
+import forestry.core.recipes.RecipeUtil;
+import forestry.core.utils.ItemStackUtil;
+import forestry.core.utils.Log;
+import forestry.core.utils.ModUtil;
+import forestry.farming.FarmRegistry;
+import forestry.farming.circuits.CircuitFarmLogic;
+import forestry.farming.logic.FarmLogicRubber;
+import forestry.farming.logic.farmables.FarmableSapling;
+import forestry.modules.ForestryModuleUids;
+import forestry.modules.ModuleHelper;
+
+import techreborn.api.ISubItemRetriever;
+import techreborn.api.TechRebornAPI;
+
+@ForestryModule(containerID = ForestryCompatPlugins.ID, moduleID = ForestryModuleUids.TECH_REBORN, name = "TechReborn", author = "temp1011", url = Constants.URL, unlocalizedDescription = "for.module.techreborn.description")
+public class PluginTechReborn extends CompatPlugin {
+
+	public static final String MOD_ID = "techreborn";
+
+	@ItemStackHolder("techreborn:rubber_log")
+	@Nullable
+	public static ItemStack RUBBER_WOOD = null;
+	@ItemStackHolder("techreborn:rubber_sapling")
+	@Nullable
+	public static final ItemStack RUBBER_SAPLING = null;
+
+	public static ItemStack sap;
+	public static ItemStack rubber;
+
+	public PluginTechReborn() {
+		super("TechReborn", MOD_ID);
+	}
+
+	@Override
+	public void preInit() {
+		IFarmProperties rubberFarm = FarmRegistry.getInstance().registerLogic("farmRubber", FarmLogicRubber::new);
+
+		Circuits.farmRubberManual = new CircuitFarmLogic("manualRubber", rubberFarm, true);
+	}
+
+	@Override
+	@Optional.Method(modid = "techreborn")
+	public void doInit() {
+		ISubItemRetriever getItem = TechRebornAPI.subItemRetriever;
+		sap = getItem.getPartByName("rubberSap");
+		rubber = getItem.getPartByName("rubber");
+
+	}
+
+	@Override
+	@Optional.Method(modid = "techreborn")
+	public void registerCrates() {
+		ItemStack scrap = TechRebornAPI.subItemRetriever.getPartByName("scrap");
+		ItemStack uuMatter = getItemStack("uumatter");
+		ModuleHelper.registerCrate(sap);
+		ModuleHelper.registerCrate(scrap);
+		ModuleHelper.registerCrate(uuMatter);
+		ModuleHelper.registerCrate("ingotLead");
+		ModuleHelper.registerCrate("ingotAluminum");
+		ModuleHelper.registerCrate("ingotBrass");
+		ModuleHelper.registerCrate("ingotNickel");
+
+		if (!ModUtil.isModLoaded(PluginIC2.MOD_ID)) {
+			ModuleHelper.registerCrate("ingotSilver");
+			ModuleHelper.registerCrate("itemRubber");
+		}
+	}
+
+	@Override
+	public void registerBackpackItems() {
+		ModuleHelper.addItemToBackpack(BackpackManager.FORESTER_UID, sap);
+	}
+
+	@Override
+	@Optional.Method(modid = "techreborn")
+	public void registerRecipes() {
+		RecipeManagers.fabricatorManager.addRecipe(ItemStack.EMPTY,
+				Fluids.GLASS.getFluid(500),
+				ModuleCore.items.tubes.get(EnumElectronTube.RUBBER, 4),
+				new Object[]{" X ", "#X#", "XXX", '#', "dustRedstone", 'X', "itemRubber"});
+
+		ItemRegistryApiculture beeItems = ModuleApiculture.getItems();
```

Comment:
Check that `ModuleApiculture` is loaded, this line will crash when the apiculture module is disabled in the config.